### PR TITLE
[Feature] Visualize openpose skeleton style on the image

### PR DESCRIPTION
**Pull request by Zheng-LinXiao**
## Use cases (Optional)
add `--skeleton-style` to `image_demo.py` and `topdown_demo_with_mmdet.py`
For Example:
```shell
python demo/topdown_demo_with_mmdet.py \
    ${MMDET_CONFIG_FILE} ${MMDET_CHECKPOINT_FILE} \
    ${MMPOSE_CONFIG_FILE} ${MMPOSE_CHECKPOINT_FILE} \
    --input ${INPUT_PATH} \
    --skeleton-style openpose \
    --show
```
The result is as follow:

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by Zheng-LinXiao** [comment]
Use 'mim install "mmpose>=1.0.0rc1"' to install mmpose, and an error will be reported when running the code. But using "pip install -v -e ." to install mmpose, running the code doesn't throw any error.

**Comment by Ben-Louis** [comment]
> Use 'mim install "mmpose>=1.0.0rc1"' to install mmpose, and an error will be reported when running the code. But using "pip install -v -e ." to install mmpose, running the code doesn't throw any error.

Thank you for your contribution! There's no need to fret about this issue. The recently introduced features won't be incorporated into mmpose when installed via pip or mim until a new release is available. However, these features are indeed present in the source code, so they will be accessible if mmpose is installed directly from the source.

**Comment by Zheng-LinXiao** [comment]
> > Use 'mim install "mmpose>=1.0.0rc1"' to install mmpose, and an error will be reported when running the code. But using "pip install -v -e ." to install mmpose, running the code doesn't throw any error.
> 
> 
> 
> Thank you for your contribution! There's no need to fret about this issue. The recently introduced features won't be incorporated into mmpose when installed via pip or mim until a new release is available. However, these features are indeed present in the source code, so they will be accessible if mmpose is installed directly from the source.

OK，thank you

File: mmpose/visualization/local_visualizer.py [hunk 1]
```diff
@@ -243,6 +248,20 @@ def _draw_instances_kpts(self,
             else:
                 keypoints_visible = [np.ones(len(kpts)) for kpts in keypoints]
 
+            if skeleton_style == 'openpose':
+                neck = (keypoints[:, 5] + keypoints[:, 6]) / 2
+                neck_score = (scores[:, 5] + scores[:, 6]) / 2
```
Comment (Tau-J) [inline mmpose/visualization/local_visualizer.py:253]: Regarding the confidence score for the `neck`, I think it should be considered based on different situations:
1. When both `left_shoulder` and `right_shoulder` are present (score >= args.kpt_thr), the average score of the two can be used as the score for the neck. 
3. When one shoulder is missing, the score for the neck should be 0

Comment (Tau-J) [reply mmpose/visualization/local_visualizer.py:253]: BTW, can you also fix this in `projects/mmpose4aigc/openpose_visualization.py` this [line](https://github.com/open-mmlab/mmpose/blob/71d74de89cdd87f147f0b091e65f07b5fe6450f0/projects/mmpose4aigc/openpose_visualization.py#L67)?

File: mmpose/visualization/local_visualizer.py [hunk 2]
```diff
@@ -243,6 +248,20 @@ def _draw_instances_kpts(self,
...
+                keypoints = np.insert(keypoints, 17, neck, axis=1)
+                scores = np.insert(scores, 17, neck_score, axis=1)
+
+                original_cols = [
+                    17, 6, 8, 10, 7, 9, 12, 14, 16, 13, 15, 2, 1, 4, 3
+                ]
+                new_cols = [1, 2, 3, 4, 6, 7, 8, 9, 10, 12, 13, 14, 15, 16, 17]
+                keypoints[:, new_cols] = keypoints[:, original_cols]
+                scores[:, new_cols] = scores[:, original_cols]
+                keypoints_visible = [np.ones(len(kpts)) for kpts in keypoints]
```
Comment (Tau-J) [inline mmpose/visualization/local_visualizer.py:263]: When visualizing images from GT in openpose-style, `keypoints_visible` should not be np.ones() for all keypoints

File: demo/image_demo.py [hunk 3]
```diff
@@ -50,8 +69,16 @@ def main():
...
 
     # init visualizer
+    model.cfg.visualizer.radius = args.radius
+    model.cfg.visualizer.alpha = args.alpha
+    model.cfg.visualizer.line_width = args.thickness
+    dataset_meta = model.dataset_meta
+    if dataset_meta.get(
+            'dataset_name') == 'coco' and args.skeleton_style == 'openpose':
+        dataset_meta = parse_pose_metainfo(
+            dict(from_file='configs/_base_/datasets/coco_openpose.py'))
```
Comment (Tau-J) [inline demo/image_demo.py:79]: Should we move this piece of code into the visualizer?

File: mmpose/visualization/local_visualizer.py (hunk 2 above)
Comment (Zheng-LinXiao) [reply mmpose/visualization/local_visualizer.py:263]: Does `keypoints_visible` to the `joint_weights` in coco_openpose.py?Here I refer to https://github.com/open-mmlab/mmpose/pull/2115/files/9630c10a895bec012cd44b2a882f8c1c143d9318#diff-070475f91f5ca244130def9db60fe354566df801e1a23aa33d3b2060c9c8515cL244,I don't understand `keypoints_visible`

Comment (Tau-J) [reply mmpose/visualization/local_visualizer.py:263]: just refer to L246-L249:
```python
if 'keypoints_visible' in instances:
    keypoints_visible = instances.keypoints_visible
else:
    keypoints_visible = [np.ones(len(kpts)) for kpts in keypoints]
```
`joint_weights` is used in loss calculation.

File: demo/image_demo.py (hunk 3 above)
Comment (Zheng-LinXiao) [reply demo/image_demo.py:79]: ok，Let me try

File: mmpose/visualization/local_visualizer.py (hunk 2 above)
Comment (Zheng-LinXiao) [reply mmpose/visualization/local_visualizer.py:263]: > When visualizing images from GT in openpose-style, `keypoints_visible` should not be np.ones() for all keypoints

Thank you very much for your explanation, I understand now!

**Review by Tau-J — APPROVED** [review]

**Pull request merged**